File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int x : nums){
            if(set.contains(x))set.remove(x);
            else set.add(x);
        }
        int a ;
        for(int i : set)
            return i;
        return -1;
    }
}

File: leetcode/0136-single-number.js
function singleNumber(nums) {
    const set = new Set();

    for (const x of nums) {
        if (set.has(x)) {
            set.delete(x);
        } else {
            set.add(x);
        }
    }

    for (const x of set) {
        return x;
    }

    return -1;
}